### PR TITLE
Add Only Authoritative Server Replication

### PR DESCRIPTION
**Pull request by nurnbeck**
#### Description
Added an auth server only replication type and updated labeled
applicable areas of the GDK.

Engine 4.26 change: https://github.com/improbableio/UnrealEngine/pull/1061
Engine 4.27 change: https://github.com/improbableio/UnrealEngine/pull/1058

#### Tests
How did you test these changes prior to submitting this pull request? In customer's codebase.
What automated tests are included in this PR?
updated SpatialGDKSchemaGeneratorTest

#### Reminders (IMPORTANT)
If your change relies on a breaking engine change:
* Increment `SPATIAL_ENGINE_VERSION` in `Engine\Source\Runtime\Launch\Resources\SpatialVersion.h` (in the engine fork) as well as `SPATIAL_GDK_VERSION` in `SpatialGDK\Source\SpatialGDK\Public\Utils\EngineVersionCheck.h`. This helps others by providing a more helpful message during compilation to make sure the GDK and the Engine are up to date.

If your change updates `Setup.bat`, `Setup.sh`, core SDK version, any C# tools in `SpatialGDK\Build\Programs\Improbable.Unreal.Scripts`, or hand-written schema in `SpatialGDK\Extras\schema`:
* Increment the number in `RequireSetup`. This will automatically run `Setup.bat` or `Setup.sh` when the GDK is next pulled.

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h [hunk 1]
```diff
@@ -39,6 +39,8 @@ FORCEINLINE ESchemaComponentType GetGroupFromCondition(ELifetimeCondition Condit
 		return SCHEMA_InitialOnly;
 	case COND_ServerOnly:
 		return SCHEMA_ServerOnly;
+	case COND_AuthServerOnly:
```
Comment (nurnbeck) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h:42]: not sure why it is complaining about this case in the include, even tried manually including the header its from and made no difference?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 2]
```diff
@@ -931,13 +931,16 @@ void ActorSystem::HandleIndividualAddComponent(const Worker_EntityId EntityId, c
 	const bool bIsServer = NetDriver->IsServer();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:931]: nit: add comment for extra server-only auth permutation

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 #include "Utils/GDKPropertyMacros.h"
 #include "Utils/SchemaDatabase.h"
-
+#include "UObject/CoreNetTypes.h"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h:9]: remove this

**Comment by m-samiec** [comment]
Please increment `ESchemaDatabaseVersion` also

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 4]
```diff
@@ -928,16 +928,20 @@ void ActorSystem::HandleIndividualAddComponent(const Worker_EntityId EntityId, c
 	// Non-owning client - data
 	// If initial-only disabled + initial-only to all (counter-intuitive, but initial only is sent as normal if disabled and not sent at all
 	// on dynamic components if enabled)
+	// Auth Server - AuthServerOnly replication, replicated between authoritative servers on authority transfer.
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:931]:
```suggestion
	// Auth Server - data/owner-only/handover/auth-server-only
```

**Review by m-samiec — APPROVED** [review]
Looks good! Could you also make an addition to `changelog.md` under the `features` section.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 5]
```diff
@@ -1181,6 +1184,13 @@ SCHEMA_GENERATOR_TEST(GIVEN_actor_class_WHEN_generating_schema_THEN_expected_com
 			{
 				return false;
 			}
+
+			// Auth Server Only expected to have not components in schema?
+			if (ComponentType == SCHEMA_AuthServerOnly)
+			{
+				return true;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:1191]: this should be a continue, not a return

Comment (nurnbeck) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:1191]: Yeah, wasn't sure of the behavior, hence the (?)! haha So then I'd have something like below, or is the set broken if it is empty?                                                
`
			if (ComponentType != SCHEMA_AuthServerOnly)
			{
				// We should have a class for each type of set
				TestTrue("Set is not empty", DataComponents->ComponentIDs.Num() > 0);
			}`

Comment (nurnbeck) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:1191]: ah, I see I missed creating a new dummy object within SchemaGenObjectstub, will make one and verify

**Pull request merged**